contracts/ColonyTask.sol

```diff
@@ -453,25 +453,27 @@ contract ColonyTask is ColonyStorage {
       role.rating = role.rateFail ? TaskRatings.Unsatisfactory : TaskRatings.Satisfactory;
     }
 
-    uint payout = task.payouts[roleId][token];
-    int reputation = getReputation(int(payout), uint8(role.rating), role.rateFail);
+    uint256 payout = task.payouts[roleId][token];
+    int256 reputation = getReputation(payout, role.rating, role.rateFail);
 
     colonyNetworkContract.appendReputationUpdateLog(role.user, reputation, domains[task.domainId].skillId);
     if (roleId == WORKER) {
       colonyNetworkContract.appendReputationUpdateLog(role.user, reputation, task.skills[0]);
     }
   }
 
-  function getReputation(int payout, uint8 rating, bool rateFail) internal pure returns(int reputation) {
-    require(rating > 0 && rating <= 3, "colony-task-rating-invalid");
+  function getReputation(uint256 payout, TaskRatings rating, bool rateFail) internal pure returns(int256) {
+    require(rating != TaskRatings.None, "colony-task-rating-invalid");
 
-    // -1, 1, 1.5 multipliers, -0.5 penalty
-    int8[3] memory ratingMultipliers = [-2, 2, 3];
-    int8 ratingDivisor = 2;
+    bool negative = (rating == TaskRatings.Unsatisfactory);
+    uint256 reputation = mul(payout, (rating == TaskRatings.Excellent) ? 3 : 2);
 
-    reputation = SafeMath.mulInt(payout, ratingMultipliers[rating - 1]);
-    reputation = SafeMath.subInt(reputation, rateFail ? payout : 0); // Deduct penalty for not rating
-    reputation /= ratingDivisor; // We may lose one atom of reputation here :sad:
+    if (rateFail) {
+      reputation = negative ? add(reputation, payout) : sub(reputation, payout);
```

Comment:
That's something fishy here.. I'm going 🎣

Reply:
I know it's not the clearest but the idea was to work with `uint`s to take advantage of the larger range, and then scale it down and add the sign at the very end.